models/spend_daily.sql

```diff
@@ -0,0 +1,193 @@
+{{ config(materialized='table') }}
+
+with dates as (
+{{ dbt_utils.date_spine(
+        datepart="day",
+        start_date="cast('2019-01-01' as date)",
+        end_date="cast('2025-01-01' as date)"
+    )
+    }}
+),
+
+dates_with_days_in_month as (
+    select
+        date_day as date,
+        day(last_day(date_day)) as days_in_month
+    from dates
+),
+
+storage_terabytes_daily as (
+    select
+        usage_date,
+        'storage' as usage_type,
+        'table and time travel' as usage_type_subcategory_1,
+        storage_bytes / power(1024, 4) as storage_terabytes
+    from snowflake.account_usage.storage_usage
+    union all
+    select
+        usage_date,
+        'storage' as usage_type,
+        'stage' as usage_type_subcategory_1,
+        stage_bytes / power(1024, 4) as storage_terabytes
+    from snowflake.account_usage.storage_usage
+    union all
+    select
+        usage_date,
+        'storage' as usage_type,
+        'failsafe' as usage_type_subcategory_1,
+        failsafe_bytes / power(1024, 4) as storage_terabyte
+    from snowflake.account_usage.storage_usage
+)
+
+, storage_spend_daily as (
+    select
+        storage_terabytes_daily.usage_date,
+        storage_terabytes_daily.usage_type,
+        storage_terabytes_daily.usage_type_subcategory_1,
+        coalesce(storage_terabytes_daily.storage_terabytes / dates_with_days_in_month.days_in_month * daily_rates.effective_rate, 0) as spend_usd
+    from storage_terabytes_daily
+    left join dates_with_days_in_month on storage_terabytes_daily.usage_date = dates_with_days_in_month.date
+    left join {{ ref('daily_rates') }} on
+        storage_terabytes_daily.usage_date = daily_rates.date
+        and daily_rates.usage_type = 'storage'
+),
+
+compute_spend_daily as (
+    select
+        metering_history.start_time::date as usage_date,
+        'warehouse compute' as usage_type,
+        metering_history.name as usage_type_subcategory_1,
+        sum(metering_history.credits_used_compute * daily_rates.effective_rate) as spend_usd
+    from snowflake.account_usage.metering_history
+    left join {{ ref('daily_rates') }} on
+        metering_history.start_time::date = daily_rates.date
+        and daily_rates.usage_type = 'compute'
+        and daily_rates.service_type = 'COMPUTE'
+    where metering_history.service_type = 'WAREHOUSE_METERING'
+    group by 1, 2, 3
+),
+
+cloud_services_spend_daily as (
```

Comment:
does this include an adjustment for the free 10%?

i'm trying to reconcile against another snowflake account, and i'm seeing ~$400 in monthly cloud services cost. i'm like 90% sure they are never charged for cloud services based on past bills i've looked at.

query i'm running to validate? 

```sql
SELECT
    DATE_TRUNC('month', usage_date) as dt,
    usage_type,
    SUM(spend_usd) AS monthly_spend
FROM spend_daily
GROUP BY 1,2
```